java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthViewVector.java

```diff
@@ -0,0 +1,1464 @@
+   * than the request one because it will round up the provided value to the nearest
+   * power of two.
+   *
+   * @param desiredAllocSize the desired new allocation size
+   * @throws OversizedAllocationException if the desired new size is more than
+   *                                      max allowed
+   * @throws OutOfMemoryException if the internal memory allocation fails
+   */
+  public void reallocViewBuffer(long desiredAllocSize) {
+    if (desiredAllocSize == 0) {
+      return;
+    }
+    long newAllocationSize = CommonUtil.nextPowerOfTwo(desiredAllocSize);
+    assert newAllocationSize >= 1;
+
+    checkDataBufferSize(newAllocationSize);
+    // for each set operation, we have to allocate 16 bytes
+    // here we are adjusting the desired allocation-based allocation size
+    // to align with the 16bytes requirement.
+    newAllocationSize += VIEW_BUFFER_SIZE;
+
+    final ArrowBuf newBuf = allocator.buffer(newAllocationSize);
+    newBuf.setBytes(0, viewBuffer, 0, viewBuffer.capacity());
+
+    viewBuffer.getReferenceManager().release();
+    viewBuffer = newBuf;
+    lastValueAllocationSizeInBytes = viewBuffer.capacity();
+  }
+
+  /**
+   * Reallocate the data buffer for reference buffer.
+   *
+   * @param desiredAllocSize allocation size in bytes
+   */
+  public void reallocViewReferenceBuffer(long desiredAllocSize) {
+    if (desiredAllocSize == 0) {
+      return;
+    }
+
+    if (dataBuffers.isEmpty()) {
+      return;
+    }
+
+    ArrowBuf currentBuf = dataBuffers.get(dataBuffers.size() - 1);
+    if (currentBuf.capacity() - currentBuf.writerIndex() >= desiredAllocSize) {
+      return;
+    }
+
+    final long newAllocationSize = CommonUtil.nextPowerOfTwo(desiredAllocSize);
+    assert newAllocationSize >= 1;
+
+    checkDataBufferSize(newAllocationSize);
+
+    final ArrowBuf newBuf = allocator.buffer(newAllocationSize);
+    dataBuffers.add(newBuf);
+  }
+
+  /**
+  *  Reallocate Validity buffer.
+  */
+  public void reallocValidityBuffer() {
+    int targetValidityCount = capAtMaxInt((validityBuffer.capacity() * 8) * 2);
+    if (targetValidityCount == 0) {
+      if (lastValueCapacity > 0) {
+        targetValidityCount = lastValueCapacity;
+      } else {
+        targetValidityCount = 2 * INITIAL_VALUE_ALLOCATION;
+      }
+    }
+
+    long validityBufferSize = computeValidityBufferSize(targetValidityCount);
+
+    final ArrowBuf newValidityBuffer = allocator.buffer(validityBufferSize);
+    newValidityBuffer.setBytes(0, validityBuffer, 0, validityBuffer.capacity());
+    newValidityBuffer.setZero(validityBuffer.capacity(), newValidityBuffer.capacity() - validityBuffer.capacity());
+    validityBuffer.getReferenceManager().release();
+    validityBuffer = newValidityBuffer;
+
+    lastValueCapacity = getValueCapacity();
+  }
+
+  private long computeValidityBufferSize(int valueCount) {
+    return (valueCount + 7) / 8;
+  }
+
+  /**
+   * Get the size (number of bytes) of underlying data buffer.
```

Comment:
Let's be consistent and call this the view buffer, given the data buffer is separate?